### ISSUE TITLE
dbt-materialize: use connection `options` instead of `SET`

### ISSUE THREAD

**Issue by morsapaes** [MaterializeInc/materialize#18263]
On connection, we set the cluster dbt should run introspection in to `mz_introspection` using:

```python
mz_introspection_cluster = "mz_introspection"
logger.debug("Switching to cluster '{}'".format(mz_introspection_cluster))
cursor.execute("SET cluster = %s" % mz_introspection_cluster)
```

After #17853, we could properly do this by passing `cluster=mz_introspection` as a connection option, instead. 

### Open question

Do we want to allow setting other options, like `search_path`? `dbt-postgres` exposes [`search_path`](https://docs.getdbt.com/reference/warehouse-setups/postgres-setup#search_path), but the implementation doesn't seem to work for Materialize (I suspect because of #18233, though it's been reported as not working for the Postgres adapter, either). I can see this being useful mostly to avoid pre-fixing connections with `{{ target.schema }}`.

**Comment by dehume** [comment]
Can this be closed after https://github.com/MaterializeInc/materialize/pull/18312?

**Comment by morsapaes** [comment]
I used `mz_introspection` as an example here, but this generalizes to the `cluster` option in `profiles.yml`. It'd be technically more correct to connect using the cluster defined as the default for the profile (we default to `default`, even if it doesn't exist), but since we fixed #17197 and #19318, there's little benefit to that. Overriding the `dbt-postgres` adapter to achieve this also seemed like a nightmare, last I looked into it, so we can close for now.
